nsv13/code/modules/munitions/ship_weapons/ballistic_weapons/pdc.dm

```diff
@@ -0,0 +1,71 @@
+/obj/machinery/ship_weapon/pdc_mount
+	name = "PDC loading rack"
+	icon = 'nsv13/icons/obj/munitions.dmi'
+	icon_state = "pdc"
+	desc = "Seegson's all-in-one PDC targeting computer, ammunition loader, and human interface has proven extremely popular in recent times. It's rare to see a ship without one of these."
+	anchored = TRUE
+	density = FALSE
+	pixel_y = 26
+
+	var/buildstage
+
+	fire_mode = FIRE_MODE_PDC
+	weapon_type = new/datum/ship_weapon/pdc_mount
+	magazine_type = /obj/item/ammo_box/magazine/pdc
+
+	auto_load = TRUE
+	semi_auto = TRUE
+	maintainable = FALSE
+	max_ammo = 100
+
+	// We're fully automatic, so just the loading sound is enough
+	mag_load_sound = 'sound/weapons/autoguninsert.ogg'
+	mag_unload_sound = 'sound/weapons/autoguninsert.ogg'
+	feeding_sound = null
+	fed_sound = null
+	chamber_sound = null
+
+	load_delay = 50
+	unload_delay = 50
+
+	// No added delay between shots or for feeding rounds
+	feed_delay = 0
+	chamber_delay_rapid = 0
+	chamber_delay = 0
+
+// Update the icon to reflect how far our magazine is depleted
+/obj/machinery/ship_weapon/pdc_mount/update_icon()
+	if(!magazine)
```

Comment:
sadly the new practise is to call parents on update_icon at TG I dont know if this codebase needs to follow suit yet. See 
https://github.com/tgstation/tgstation/pulls?utf8=%E2%9C%93&q=is%3Apr+is%3Aclosed+label%3A%22Code+Improvement%22+icon
This might be a few months or more away it depends on how bee handles this.